pkg/registry/storage/volumeattachment/strategy.go

```diff
@@ -72,3 +102,30 @@ func (volumeAttachmentStrategy) ValidateUpdate(ctx context.Context, obj, old run
 func (volumeAttachmentStrategy) AllowUnconditionalUpdate() bool {
 	return false
 }
+
+// volumeAttachmentStatusStrategy implements behavior for VolumeAttachmentStatus subresource
+type volumeAttachmentStatusStrategy struct {
+	volumeAttachmentStrategy
+}
+
+// StatusStrategy is the default logic that applies when creating and updating
+// VolumeAttachmentStatus subresource via the REST API.
+var StatusStrategy = volumeAttachmentStatusStrategy{Strategy}
+
+// PrepareForUpdate sets the Status fields which is not allowed to be set by an end user updating a VolumeAttachment
+func (volumeAttachmentStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newVolumeAttachment := obj.(*storage.VolumeAttachment)
+	oldVolumeAttachment := old.(*storage.VolumeAttachment)
+
+	newVolumeAttachment.Spec = oldVolumeAttachment.Spec
+
+	oldMeta := oldVolumeAttachment.ObjectMeta
+	newMeta := &newVolumeAttachment.ObjectMeta
+	newMeta.SetDeletionTimestamp(oldMeta.GetDeletionTimestamp())
```

Comment:
This is a lot of fields to remember to reset for probably every api type, and a lot of places to update if we add new fields to meta in the future.  Could we have some common "resetMeta" function that all api types could share?

Reply:
I would hope that a variant of commit that @liggitt linked will be merged in future in api-machinery repo but I wanted to avoid scope of this PR to not include apimachinery too. But I can indeed move this to a common util function.

Reply:
Agree I think this could be done as a separate cleanup pr

Reply:
logged here - https://github.com/kubernetes/kubernetes/issues/71056